File: totosuki/ABC/C/006.py
import sys
input = sys.stdin.buffer.readline

N, M = map(int, input().split())

for i in range(N+1):
  for j in [0,1]:
    k = N - (i + j)
    if (i*2 + j*3 + k*4) == M and k >= 0:
      print(i, j, k)
      break
  else:
    continue
  break
else:
  print(-1, -1, -1)

# for i in range(N+1):
#   for j in range(N+1-i):
#     k = N - (i + j)
#     if (i*2 + j*3 + k*4) == M:
#       print(i, j, k)
#       break
#   else:
#     continue
#   break
# else:
#   print(-1, -1, -1)

File: OCH_Tao/ABC/A/006.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").split("\n");
const N = parseInt(input[0], 10);
if (input[0].includes("3")) {
  console.log("YES");
} else if (N % 3 === 0) {
  console.log("YES");
} else {
  console.log("NO");
}